src/NuGet.Core/NuGet.Commands/PackCommandRunner.cs

```diff
@@ -244,26 +245,38 @@ private static void AddDependencyGroups(IList<LibraryDependency> dependencies, N
                     continue;
                 }
 
-                List<string> includes = new List<string>();
-                if (effectiveInclude == LibraryIncludeFlags.All)
+                if (isTargetFrameworks && dependency.LibraryRange.TypeConstraint == LibraryDependencyTarget.Project)
```

Comment:
This should be treated the same regardless of if it appears at the root or under a framework. I suggest combining the root level dependencies with each framework specific group.

It is true that framework references can only appear under the framework, but that should get handled by the type check below.